ingestion/moj_statistical_publications_source/source.py

```diff
@@ -0,0 +1,298 @@
+import datetime
+import logging
+from io import BufferedReader
+from typing import Any, Dict, Generator, Iterable, List, Optional
+from urllib.parse import urljoin
+
+import datahub.emitter.mce_builder as mce_builder
+import datahub.emitter.mcp_builder as mcp_builder
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.decorators import (
+    SupportStatus,
+    config_class,
+    platform_name,
+    support_status,
+)
+from datahub.ingestion.api.source import (
+    CapabilityReport,
+    SourceReport,
+    TestableSource,
+    TestConnectionReport,
+)
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.common.subtypes import DatasetContainerSubTypes
+from datahub.metadata.com.linkedin.pegasus2avro.common import TimeStamp
+from datahub.metadata.schema_classes import (
+    ContainerClass,
+    DataPlatformInstanceClass,
+    DatasetPropertiesClass,
+    DomainsClass,
+    GlobalTagsClass,
+    SubTypesClass,
+    TagAssociationClass,
+)
+from datahub.utilities.time import datetime_to_ts_millis
+
+from ingestion.ingestion_utils import FindMojDataEntityTypes, format_domain_name
+
+from .api_client import MojPublicationsAPIClient
+from .config import MojPublicationsAPIConfig
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+@platform_name("File")
+@config_class(MojPublicationsAPIConfig)
+@support_status(SupportStatus.CERTIFIED)
+class MojPublicationsAPISource(TestableSource):
+    """
+    This plugin pulls metadata from the gov.uk publications search and content APIs
+    """
+
+    def __init__(
+        self,
+        ctx: PipelineContext,
+        config: MojPublicationsAPIConfig,
+        validate_domains: bool = True,
+    ) -> None:
+        self.ctx = ctx
+        self.config = config
+        self.report = SourceReport()
+        self.fp: Optional[BufferedReader] = None
+        self.client = MojPublicationsAPIClient(
+            config.base_url, config.default_contact_email, config.params
+        )
+        if validate_domains:
+            self.client.validate_domains()
```

Comment:
Is this purely for testing? don't we always want to validate that the domains are ok? A mock in the tests could help with the testing

Reply:
nah this is to check we've not tried to add a domain in the mapping file that does not exist in datahub

Reply:
I mean why is it in the if statement?

Reply:
good question...